packages/blockchain-wallet-v4-frontend/src/middleware/analyticsMiddleware.ts

```diff
@@ -195,14 +195,26 @@ const simpleBuyPaymentTypeDictionary = (rawPaymentType: SBPaymentTypes) => {
 
 const getOriginalTimestamp = () => new Date().toISOString()
 
+const id = v4()
+
 const analytics = queuevent<AnalyticsKey, AnalyticsPayload>({
   queueCallback: async (rawEvents) => {
     const res = await fetch('/wallet-options-v4.json')
     const options = await res.json()
 
     const analyticsURL = `${options.domains.api}/events/publish`
 
-    const id = v4()
+    const context: { traits: { nabu_id: string | null } } = {
+      traits: {
+        nabu_id: null
+      }
+    }
+
+    const nabuId = rawEvents.find((event) => event.payload.id)?.payload.id
+
+    if (nabuId) {
+      context.traits.nabu_id = nabuId
```

Comment:
you could default nabuId to null on line 213 and then simply remove the `if` logic here.